substrate/client/rpc-servers/src/lib.rs

```diff
@@ -153,7 +170,10 @@ pub async fn start_server<M: Send + Sync + 'static>(
 				let transport_label = if is_websocket { "ws" } else { "http" };
 
 				let metrics = metrics.map(|m| MetricsLayer::new(m, transport_label));
-				let rpc_middleware = RpcServiceBuilder::new().option_layer(metrics.clone());
+				let rate_limit = rate_limit.map(|r| RateLimitLayer::per_minute(r));
+				let rpc_middleware =
+					RpcServiceBuilder::new().option_layer(rate_limit).option_layer(metrics.clone());
```

Comment:
Do we want to add rate-limited methods to the call metrics? I think it might still be useful to report them, wdyt?

If so, we should register the `metrics` as an optional layer first.

Reply:
yupp, good catch